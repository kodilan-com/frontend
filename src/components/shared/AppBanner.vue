<script>
import queryUtils from '../../utils/query';
import LocationSelect from './LocationSelect';

export default {
  components: {
    LocationSelect,
  },
  data() {
    return {
      params: {
        query: '',
        location: '',
      },
    };
  },
  methods: {
    handleSearch() {
      const params = queryUtils.getParams(this.params);

      if (params) {
        this.$router.push({ path: 'ilan-ara', query: params });
      }
    },
  },
};
</script>

<template>
  <div id="banner">
    <div class="container">
      <div class="sixteen columns">
        <div class="search-container">
          <h2>İlan Ara</h2>
          <input
            v-model="params.query"
            @keypress.enter="handleSearch"
            type="text"
            class="ico-01"
            placeholder="Pozisyon adı, teknoloji adı"
          >
          <div class="banner--location-select">
            <location-select
              v-model="params.location"
              :value="params.location"
              :searchable="true"
            />
          </div>
          <button @click="handleSearch" class="tag-search-btn">
            <i class="fa fa-search tag-search-btn" />
          </button>
          <div class="clearfix" />
          <div class="browse-jobs">
            <router-link to="/etiket/frontend" class="tag-header-tags">
              Frontend
            </router-link>
            <router-link to="/etiket/backend" class="tag-header-tags">
              Backend
            </router-link>
            <router-link to="/etiket/mobile" class="tag-header-tags">
              Mobile
            </router-link>
            <router-link to="/etiket/devops" class="tag-header-tags">
              DevOps
            </router-link>
          </div>
          <div class="browse-jobs next-line">
            <router-link to="/ilan-ara?location=istanbul" class="tag-header-tags">
              İstanbul
            </router-link>
            <router-link to="/ilan-ara?location=izmir" class="tag-header-tags">
              İzmir
            </router-link>
            <router-link to="/ilan-ara?location=ankara" class="tag-header-tags">
              Ankara
            </router-link>
            <router-link to="/ilan-ara?location=Remote" class="tag-header-tags">
              Remote
            </router-link>
          </div>
        </div>
      </div>
    </div>
  </div>
</template>
