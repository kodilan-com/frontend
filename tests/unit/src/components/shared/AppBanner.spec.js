import { shallowMount, createLocalVue } from '@vue/test-utils';
import AppBanner from '@/components/shared/AppBanner.vue';
import queryUtils from '@/utils/query';

describe('AppBanner.vue', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  describe('data', () => {
    it('should have "query" and "location" properties', () => {
      const defaultComponentState = {
        params: {
          query: '',
          location: ''
        }
      };
      const componentState = AppBanner.data();

      expect(defaultComponentState).toEqual(componentState);
    })
  });

  describe('methods', () => {
    describe('handleSearch', () => {
      let wrapper;

      beforeEach(() => {
        wrapper = shallowMount(AppBanner, {
          stubs: ['router-link'],
          mocks: {
            $router: {
              push: jest.fn()
            }
          }
        });
      });

      it('should not push to router the "ilan-ara" path, if "params" is false', () => {
        queryUtils.getParams = jest.fn(() => false);

        wrapper.vm.handleSearch();
        queryUtils.getParams.mockRestore()
        expect(wrapper.vm.$router.push).not.toHaveBeenCalled();
      })

      it('should push to router the "ilan-ara" path with "test-param" query string, if "params" is return "test-param"', () => {
        queryUtils.getParams = jest.fn(() => "test-params");

        wrapper.vm.handleSearch();

        expect(wrapper.vm.$router.push).toHaveBeenCalledTimes(1);
        expect(wrapper.vm.$router.push).toBeCalledWith({
          path: 'ilan-ara',
          query: 'test-params'
        });
      })
    })
  });

  describe('template', () => {
    let wrapper;

    beforeEach(() => {
      wrapper = shallowMount(AppBanner, {
        stubs: ['router-link'],
      });
    });

    it('should match the snapshot', () => {
      expect(wrapper.vm.$el).toMatchSnapshot();
    });

    it('should execute "handleSearch" method if enter is pressed in searchbox', () => {
      wrapper.setMethods({ handleSearch: jest.fn() })
      
      wrapper.find('input.ico-01').trigger('keypress.enter');

      expect(wrapper.vm.handleSearch).toHaveBeenCalledTimes(1);
    });

    it('should execute "handleSearch" method if click the search button', () => {
      wrapper.setMethods({ handleSearch: jest.fn() })

      wrapper.find('button.tag-search-btn').trigger('click');

      expect(wrapper.vm.handleSearch).toHaveBeenCalledTimes(1);
    });
  });
})
