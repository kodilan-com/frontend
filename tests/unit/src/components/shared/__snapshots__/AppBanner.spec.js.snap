// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppBanner.vue template should match the snapshot 1`] = `
<div
  id="banner"
>
  <div
    class="container"
  >
    <div
      class="sixteen columns"
    >
      <div
        class="search-container"
      >
        <h2>
          İlan Ara
        </h2>
         
        <input
          class="ico-01"
          placeholder="Pozisyon adı, teknoloji adı"
          type="text"
        />
         
        <div
          class="banner--location-select"
        >
          <location-select-stub
            value=""
          />
        </div>
         
        <button
          class="tag-search-btn"
        >
          <i
            class="fa fa-search tag-search-btn"
          />
        </button>
         
        <div
          class="clearfix"
        />
         
        <div
          class="browse-jobs"
        >
          <router-link-stub
            class="tag-header-tags"
            to="/etiket/frontend"
          >
            
            Frontend
          
          </router-link-stub>
           
          <router-link-stub
            class="tag-header-tags"
            to="/etiket/backend"
          >
            
            Backend
          
          </router-link-stub>
           
          <router-link-stub
            class="tag-header-tags"
            to="/etiket/mobile"
          >
            
            Mobile
          
          </router-link-stub>
           
          <router-link-stub
            class="tag-header-tags"
            to="/etiket/devops"
          >
            
            DevOps
          
          </router-link-stub>
        </div>
         
        <div
          class="browse-jobs next-line"
        >
          <router-link-stub
            class="tag-header-tags"
            to="/ilan-ara?location=istanbul"
          >
            
            İstanbul
          
          </router-link-stub>
           
          <router-link-stub
            class="tag-header-tags"
            to="/ilan-ara?location=izmir"
          >
            
            İzmir
          
          </router-link-stub>
           
          <router-link-stub
            class="tag-header-tags"
            to="/ilan-ara?location=ankara"
          >
            
            Ankara
          
          </router-link-stub>
           
          <router-link-stub
            class="tag-header-tags"
            to="/ilan-ara?location=Remote"
          >
            
            Remote
          
          </router-link-stub>
        </div>
      </div>
    </div>
  </div>
</div>
`;
